compiler/rustc_expand/src/proc_macro_server.rs

```diff
@@ -421,13 +427,58 @@ impl server::TokenStream for Rustc<'_> {
         parse_stream_from_source_str(
             FileName::proc_macro_source_code(src),
             src.to_string(),
-            self.sess,
+            self.sess(),
             Some(self.call_site),
         )
     }
     fn to_string(&mut self, stream: &Self::TokenStream) -> String {
         pprust::tts_to_string(stream)
     }
+    fn expand_literal(&mut self, stream: &Self::TokenStream) -> Result<Self::Literal, ()> {
+        // Parse the expression from our tokenstream.
+        let expr = {
+            let mut p = rustc_parse::stream_to_parser(
+                self.sess(),
+                stream.clone(),
+                Some("proc_macro expand literal"),
+            );
+            let expr = p.parse_expr().map_err(|_| ())?;
+            if p.token != token::Eof {
+                return Err(());
+            }
+            expr
+        };
+
+        // Perform eager expansion on the expression.
+        let expr = self
+            .ecx
+            .expander()
+            .fully_expand_fragment(crate::expand::AstFragment::Expr(expr))
+            .make_expr();
+
+        // Extract the literal value from the expression.
+        let lit = match &expr.kind {
+            ast::ExprKind::Lit(l) => l.kind.to_lit_token(),
```

Comment:
```suggestion
            ast::ExprKind::Lit(l) => l.token,
```
`to_lit_token` is a best-effort recovery used mostly for pretty-printing, but we need to preserve the token precisely here.

Reply:
All the stuff like `// After expansion, all integer literals are decimal` should disappear from the tests after this fix.

Reply:
Ah... I don't how how I missed that member existing, my bad.